tests/core/test_configuration.py

```diff
@@ -596,26 +597,27 @@ def test_config_value_types(self, key, type):
         assert isinstance(section_dict[key], type)
 
     def test_auth_backends_adds_session(self):
-        test_conf = AirflowConfigParser(default_config="")
-        # Guarantee we have deprecated settings, so we test the deprecation
-        # lookup even if we remove this explicit fallback
-        test_conf.deprecated_values = {
-            "api": {
-                "auth_backends": (
-                    re.compile(r"^airflow\.api\.auth\.backend\.deny_all$|^$"),
-                    "airflow.api.auth.backend.session",
-                    "3.0",
-                ),
-            },
-        }
-        test_conf.read_dict({"api": {"auth_backends": "airflow.api.auth.backend.basic_auth"}})
+        with patch("os.environ", {"AIRFLOW__API__AUTH_BACKEND": None}):
```

Comment:
This test worked differently depending if the AUTH_BACKEND was set or not (and I had it set in breeze in init.sh to test python client. clearing the setting makes it side-effect free.